websites/F/Foxford/presence.ts

```diff
@@ -0,0 +1,66 @@
+const presence = new Presence({
+	clientId: "1019291803322679456",
+});
+
+function pathHas(pathName: string): boolean {
+	return document.location.pathname.toLowerCase().includes(pathName);
+}
+
+const routes = {
+	conspects: "Просматривает конспект",
+	account: "Настраивает профиль",
+	externship: "Просматривает программу обучения",
+	calendar: "Смотрит календарь занятий",
+	"daily-plan": "Смотрит план занятий",
+	objectives: "Просматривает задания",
+	progress: "Просматривает успеваемость",
+	dashboard: "Просматривает курсы",
+} as const;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "f1024",
+	};
+
+	switch (true) {
+		case location.pathname.split("/").pop() in routes:
+			presenceData.details =
+				routes[
+					location.pathname.split("/").pop() as unknown as keyof typeof routes
```

Comment:
I don't think the unknown type needs to be here

Reply:
Without type extension, the compiler starts to worry that you are doing something wrong

Reply:
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions